debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpModuleCommand.java

```diff
@@ -0,0 +1,256 @@
+import com.ibm.j9ddr.vm29.pointer.generated.J9HashTablePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ModulePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9PackagePointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ClassHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.JavaVersionHelper;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.ClassIteratorFilter;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.ClassOutput;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.ModuleIteratorFilter;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.ModuleOutput;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.PackageIteratorFilter;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.PackageOutput;
+
+public class DumpModuleCommand extends Command 
+{
+	enum Subcommand {
+		CLASS, MODULE, PACKAGE, ALL, HELP, INVALID
+	}
+
+	public DumpModuleCommand()
+	{
+		addCommand("dumpmodule", "[all|requires|exports|classes|packages] <moduleAddress>|help", "List details about a target module");
+	}
+	
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException 
+	{
+		Subcommand subcommand = Subcommand.INVALID;
+		ClassIteratorFilter classFilter = null;
+		ClassOutput classOutput = null;
+		ModuleIteratorFilter moduleFilter = null;
+		ModuleOutput moduleOutputter = null;
+		PackageIteratorFilter packageFilter = null;
+		PackageOutput packageOutputter = null;
+		String filterArg = null;
+		switch (args.length) {
+		case 1:
+			if (args[0].equalsIgnoreCase("help")) {
+				subcommand = Subcommand.HELP;
+			} else {
+				filterArg = args[0];
+				subcommand = Subcommand.ALL;
+			} /* else stay subcommand.INVALID */
+			break;
+		case 2:
+			filterArg = args[1];
+			switch (args[0]) {
+			case "all":
+				subcommand = Subcommand.ALL;
+				break;
+			case "requires":
+				moduleFilter = DumpModuleCommand::filterModuleRequired;
+				moduleOutputter = ModularityHelper::printJ9Module;
+				subcommand = Subcommand.MODULE;
+				break;
+			case "exports":
+				packageFilter = DumpModuleCommand::filterPackageModuleAndExport;
+				packageOutputter = ModularityHelper::printPackageExportVerbose;
+				subcommand = Subcommand.PACKAGE;
+				break;
+			case "classes":
+				classFilter = DumpModuleCommand::filterClassByModule;
+				classOutput = ModularityHelper::printJ9Class;
+				subcommand = Subcommand.CLASS;
+				break;
+			case "packages":
+				packageFilter = DumpModuleCommand::filterPackageModule;
+				packageOutputter = ModularityHelper::printJ9Package;
+				subcommand = Subcommand.PACKAGE;
+				break;
+			default:
+				break; /* default stay subcommand.INVALID */
```

Comment:
This comment should refer to `Subcommand` (with a capital 'S').
A better strategy might be to make `subcommand` 'final' and set it on each branch in place of the comments.

Reply:
I will implement `final` here and on findmodules then

Reply:
Implemented and tested on dumpmodule